plutus-core/plutus-core/src/PlutusCore/Error.hs

```diff
@@ -111,8 +120,30 @@ data Error uni fun ann
 makeClassyPrisms ''Error
 deriving stock instance (Show fun, Show ann, Closed uni, Everywhere uni Show, GShow uni, Show ParserError) => Show (Error uni fun ann)
 
-instance AsParserError (Error uni fun ann) where
-    _ParserError = _ParseErrorE
+-- To be able to carry megaparsec's @ParseErrorBundle@ straight instead of the wrapper @ParserErrorBundle@:
```

Comment:
I wouldn't worry about this. The only cost is that you have to do the newtype wrapper at the (single!) place where you throw such an error. Writing this prism stuff is IMO more work than just doing that.